homeassistant/components/somfy/__init__.py

```diff
@@ -0,0 +1,186 @@
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+API = 'api'
+
+DEVICES = 'devices'
+
+REQUIREMENTS = ['pymfy==0.4.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+DOMAIN = 'somfy'
+
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+
+NOTIFICATION_CB_ID = 'somfy_cb_notification'
+NOTIFICATION_OK_ID = 'somfy_ok_notification'
+NOTIFICATION_TITLE = 'Somfy Setup'
+
+ATTR_ACCESS_TOKEN = 'access_token'
+ATTR_REFRESH_TOKEN = 'refresh_token'
+ATTR_CLIENT_ID = 'client_id'
+ATTR_CLIENT_SECRET = 'client_secret'
+
+SOMFY_AUTH_CALLBACK_PATH = '/auth/somfy/callback'
+SOMFY_AUTH_START = '/auth/somfy'
+
+DEFAULT_CACHE_PATH = '.somfy'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+SOMFY_COMPONENTS = ['cover']
+
+
+def setup(hass, config):
+    """Set up the Somfy component."""
+    from pymfy.api.somfy_api import SomfyApi
+
+    hass.data[DOMAIN] = {}
+    is_ready = False
+
+    # This is called to create the redirect so the user can Authorize Home .
+    redirect_uri = '{}{}'.format(
+        hass.config.api.base_url, SOMFY_AUTH_CALLBACK_PATH)
+    conf = config[DOMAIN]
+    api = SomfyApi(conf.get(CONF_CLIENT_ID),
+                   conf.get(CONF_CLIENT_SECRET),
+                   redirect_uri, hass.config.path(DEFAULT_CACHE_PATH))
+    hass.data[DOMAIN][API] = api
+
+    if not api.token:
+        authorization_url, _ = api.get_authorization_url()
+        hass.components.persistent_notification.create(
+            'In order to authorize Home Assistant to view your Somfy devices'
+            ' you must visit this <a href="{}" target="_blank">link</a>.'
+            .format(authorization_url),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_CB_ID
+        )
+        hass.http.register_view(SomfyAuthCallbackView(config))
+        is_ready = True
+    else:
+        if update_all_devices(hass):
+            is_ready = True
+            for component in SOMFY_COMPONENTS:
+                discovery.load_platform(hass, component, DOMAIN, {}, config)
+
+    return is_ready
+
+
+class SomfyAuthCallbackView(HomeAssistantView):
+    """Handle OAuth finish callback requests."""
+
+    url = SOMFY_AUTH_CALLBACK_PATH
+    name = 'auth:somfy:callback'
+    requires_auth = False
+
+    def __init__(self, config):
+        """Initialize the OAuth callback view."""
+        self.config = config
+
+    @callback
+    def get(self, request):
+        """Finish OAuth callback request."""
+        from aiohttp import web
+        from oauthlib.oauth2 import MismatchingStateError
+        from oauthlib.oauth2 import InsecureTransportError
```

Comment:
these must go into REQUIREMENTS

Reply:
oauthlib is [included](https://github.com/tetienne/somfy-open-api/blob/master/setup.py#L23) within pymfy. Do you want I add it explicitly?

Reply:
thats OK then, just making sure a new install will not miss any of the required libraries

Reply:
Several person I've already test it and it was ok.